libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -27,25 +28,74 @@ CGroupsParser::CGroupsParser()
 {
 }
 
-ReturnCode CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element)
+std::string CGroupsParser::convertToBytes(const std::string settingValue, const int multiply)
+{
+    char *end;
+    long long limit_in_bytes = std::strtoll(settingValue.c_str(), &end, 10);
+    if (limit_in_bytes ==  std::numeric_limits<long long>::max() ||
+        limit_in_bytes ==  std::numeric_limits<long long>::min()) {
+        if (errno == ERANGE) {
+            log_error() << settingValue << " is out of range";
+            throw LimitRangeError(settingValue + " is out of range");
+        }
+    }
+
+    if (limit_in_bytes > llrint(std::numeric_limits<long long>::max() / multiply)) {
+        log_error() << settingValue << " * " << std::to_string(multiply) <<" is out of range";
+        throw LimitRangeError();
+    }
+
+    limit_in_bytes *= multiply;
+    return std::to_string(limit_in_bytes);
+}
+
+std::string CGroupsParser::suffixCorrection(const std::string settingValue)
+{
+    auto suffix = settingValue.back();
+    std::string removeUnit = settingValue;
+    removeUnit.pop_back();
+
+    if (std::isdigit(suffix)) {
+        //the value is digit already no need to do anything
+        return settingValue;
+    } else if ('K' == suffix || 'k' == suffix) {
+        //the value is Kilobyte unit convert it to byte
+        int KB = 1024;
+        return convertToBytes(removeUnit, KB);
+    } else if ('M' == suffix || 'm' == suffix) {
+        //the value is Megabyte unit convert it to byte
+        int MB = 1024 * 1024;
+        return convertToBytes(removeUnit, MB);
+    } else if ('G' == suffix || 'g' == suffix) {
+        //the value is Gigabyte unit convert it to byte
+        int GB = 1024 * 1024 * 1024;
+        return convertToBytes(removeUnit, GB);
+    }
+    log_error() << "Bad suffix on setting value " << settingValue;
+    throw BadSuffixError("Bad suffix on setting value " + settingValue);
```

Comment:
Consider creating a variable holding the message, since you're both logging and throwing the same message. That goes for all places in the code.